File: MAStaff-Common/src/main/java/es/angelillo15/mast/utils/StaffChatManager.java
package es.angelillo15.mast.utils;

import java.util.ArrayList;

public class StaffChatManager {
    private static final ArrayList<String> staffChatEnable = new ArrayList<>();

    public static void setStaffChatEnable(String playerUUID, boolean staffChat) {
        if (staffChat) {
            staffChatEnable.add(playerUUID);
        } else {
            staffChatEnable.remove(playerUUID);
        }
    }

    public static boolean isStaffChatEnable(String playerUUID) {
        return staffChatEnable.contains(playerUUID);
    }

    public static void removePlayer(String playerUUID) {
        staffChatEnable.remove(playerUUID);
    }

    public static void clear() {
        staffChatEnable.clear();
    }
}

File: MAStaff-API/src/main/java/es/angelillo15/mast/api/managers/StaffChatManager.kt
package es.angelillo15.mast.api.managers

import com.google.inject.Singleton
import com.nookure.mast.api.cmd.sender.CommandSender
import es.angelillo15.mast.api.config.common.CommonMessages.StaffChat.disabled
import es.angelillo15.mast.api.config.common.CommonMessages.StaffChat.enabled

@Singleton
class StaffChatManager {
  private val staffChatEnabled = ArrayList<String>()

  /**
   * @param player The player to check
   * @return If the player has the staff chat enabled
   */
  fun isStaffChatEnabled(player: String): Boolean {
    return staffChatEnabled.contains(player)
  }

  /**
   * @param player The player to set
   * @param enabled If the player has the staff chat enabled
   */
  fun setStaffChatEnabled(player: String, enabled: Boolean) {
    if (enabled) {
      staffChatEnabled.add(player)
    } else {
      staffChatEnabled.remove(player)
    }
  }

  fun toggleStaffChat(player: String) {
    val toggle = isStaffChatEnabled(player)

    setStaffChatEnabled(player, !toggle)
  }

  fun toggleAndAdvert(sender: CommandSender) {
    if (isStaffChatEnabled(sender.getName())) {
      setStaffChatEnabled(sender.getName(), false)
      sender.sendMessage(disabled())
    } else {
      setStaffChatEnabled(sender.getName(), true)
      sender.sendMessage(enabled())
    }
  }
}